stubs/stripe/stripe/api_resources/payment_intent.pyi

```diff
@@ -9,4 +11,11 @@ class PaymentIntent(CreateableAPIResource, ListableAPIResource, UpdateableAPIRes
     OBJECT_NAME: str
     def cancel(self, idempotency_key: str | None = ..., **params): ...
     def capture(self, idempotency_key: str | None = ..., **params): ...
-    def confirm(self, idempotency_key: str | None = ..., **params): ...
+    @overload
+    @classmethod
+    def confirm(
+        cls, intent: str, api_key: str | None = ..., stripe_version: str | None = ..., stripe_account: str | None = ..., **params
+    ): ...
+    @overload
+    @classmethod
+    def confirm(cls, idempotency_key: str | None = ..., **params): ...
```

Comment:
Shouldn't the second overload not be decorated with `@classmethod`? If I understand the `@class_method_variant` decorator correctly, this can be called in two versions:

```python
PaymentIntent().confirm("key", ...)
PaymentIntent.confirm("payment intent", "key", ...)
```

Reply:
This is mainly for avoiding mypy complaining about one of the methods not being decorated consistently with `@classmethod`. Another reason is that mypy does not correctly remove the first argument of the second variant without them both having `@classmethod`.

Reply:
If it works, I'm fine with this. I think it would be a good idea to add a test case for a complicated case like this, though. Maybe @AlexWaygood can help with this.

We could also try to annotate `class_method_variant()` correctly and use the decorator here. Descriptor protocol support in type checkers is decent. But that's probably out of scope for this PR.

Reply:
> I think it would be a good idea to add a test case for a complicated case like this, though. Maybe @AlexWaygood can help with this.

We don't (yet) have the infrastructure set up to add test cases for our third-party stub -- only our stdlib stubs. This has come up a few times now, though, so I can work on setting something up. But we probably shouldn't wait on merging this PR before then.

Reply:
I filed https://github.com/python/typeshed/issues/8505 as a reminder to myself.

Reply:
> We could also try to annotate class_method_variant() correctly and use the decorator here. Descriptor protocol support in type checkers is decent. But that's probably out of scope for this PR.
Yes. That's helpful for sweeping up other similar use cases of the decorator in the stubs, but I am not sure how that will be possible with type annotation alone. We can figure this out later.